core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -120,10 +141,14 @@ case class GroupSummary(state: String,
  *  3. leader id
  */
 @nonthreadsafe
-private[coordinator] class GroupMetadata(val groupId: String, val protocolType: String) {
+private[coordinator] class GroupMetadata(val groupId: String) {
 
+  private var state: GroupState = Empty
   private val members = new mutable.HashMap[String, MemberMetadata]
-  private var state: GroupState = Stable
+  private val offsets = new mutable.HashMap[TopicPartition, OffsetAndMetadata]
+  private val pendingOffsetCommits = new mutable.HashMap[TopicPartition, OffsetAndMetadata]
+
+  var protocolType: Option[String] = None
```

Comment:
Yes, that was the main case I had in mind. One alternative might be to set the protocol type as "consumer" by default, but at the moment, the protocol type definition is maintained only on the client. It's also possible prior to this patch to change the protocol type of a group by shutting down all the members and restarting them with a different protocol type. I'm not sure there's a strong reason to preserve this, but I thought we may as well.